### DISCUSSION TITLE
Slight confusion with amp= vs amplitude=... in make_params()

### DISCUSSION THREAD

**Discussion by PennyWieser** [lmfit/lmfit-py#892] [General]
Hi, First off, I love lmfit so much!
I just upgraded from 1.0.2 to 1.2.0, and I notice the syntax for
 params = model.make_params() has changed
it used to take amplitude= as an arguement, now it wants amp=..
Similarly, center= has gone to cen=

But when I then print the parameters I have made, it is called amplitude and center again. 
I couldnt seem to find any documentation explaining this syntax difference/change and I was just wondering if its intentional, or whether this is an old fashioned away of making parameters I shouldnt be using any more?

Thanks so much!

**Comment by newville** [comment]
@PennyWieser sorry for being dense, but where did the syntax change? 

Basically, the arguments to `Model.make_params()` will depend on the keywords used by the model.   So, if a model function uses `amp`, that is what `model.make_params()` would want.

**Comment by PennyWieser** [comment]
Sorry, so in version v1.0.2, you can do this:

`
#!pip install lmfit==1.0.2

import lmfit as l

from lmfit.models import GaussianModel, VoigtModel, LinearModel, ConstantModel, PseudoVoigtModel

model0 = VoigtModel(prefix='p0_')

pars0 = model0.make_params(p0_center=5)
`

But in v.1.2.0  I get this error
![image](https://github.com/lmfit/lmfit-py/assets/65668673/b9ca5ac9-1613-4e4d-abd5-94a5055c00d9)

And it doesnt error out if I do 
'pars0 = model0.make_params(p0_cent=5)`
But it also doesnt use that value I have given it. 

Does it now want you to do
params=model0.make_params()
then
params['...'].set(value)

**Comment by newville** [comment]
@PennyWieser OK, thanks, that helps clarify the problem. 

I think this is a real bug.  I'm sort of surprised we don't have a test or example for that!   
It should work to do  `pars0 = model0.make_params(center=5)`  -- for a "simple, non-composite model", that will work, but the more explicit usage with `p0_center` should work too!

I also think that the fix here is easy. If you're up for it, I think the hotfix of changing that Line 697 to  `setpar(par, kwargs[name])` will fix the problem.   

We'll have to add tests for this and then get a fix in, and then push out 1.2.2 soon.

**Reply by reneeotten** [reply]
it looks like this mistake was introduced when the `setpar` function was added as originally it read:
```par.value = kwargs[name]``` this was changed to ```setpar(par, kwargs[basename])```. 

According to our `Codecov` statistics these lines are supposedly covered in the tests, why that didn't catch this issues is indeed a good question ;)

**Reply by newville** [reply]
@reneeotten I'm not 100% certain, but I think maybe this works correctly when part of a CompositeModel, but not when a simple Model.   I'll work on tests and fix tomorrow,

**Reply by PennyWieser** [reply]
A related bug  I think, in v 1.0.2 I used to be able to do this when iteratively fitting 2 peaks to stabilize the 2nd peak
        result.params['p2_center'].vary = False
        result.params['p2_amplitude'].vary = False
        result.params['p2_sigma'].vary = False
When I run the same code in v.1.2.0, it acts the same as commenting these lines out in v.1.0.2 which is making me think this is no longer working to change the parameters? 
I tried p2_cent but got a key error. 

Is this way of fixing parameters still supported in v.1.2.0?

**Reply by newville** [reply]
Fixing parameters is definitely still supported.  
Please send a complete, minimal example that shows the problem.
